contracts/token/TokenHolder.sol

```diff
@@ -264,25 +273,24 @@ contract TokenHolder is MasterCopyNonUpgradable
         onlyOwner
         keyIsAuthorized(_sessionKey)
     {
-        sessionKeys[_sessionKey].status = AuthorizationStatus.REVOKED;
+        sessionKeys[_sessionKey].session = uint256(AuthorizationStatus.REVOKED);
 
         emit SessionRevoked(_sessionKey);
     }
 
     /**
-     * @notice Logout session of the msg.sender.
+     * @notice Logout all authorized sessions.
      *
-     * @dev Function revokes the key even if it has expired.
-     *      Function requires:
-     *          - The session key (msg.sender) is authorized.
+     * @dev Function requires:
+     *          - Only owner is allowed to call.
      */
     function logout()
         external
-        keyIsAuthorized(msg.sender)
+        onlyOwner
     {
-        sessionKeys[msg.sender].status = AuthorizationStatus.REVOKED;
+        sessionWindow = sessionWindow.add(1);
 
-        emit SessionRevoked(msg.sender);
+        emit SessionsLoggedOut(sessionWindow.sub(1));
```

Comment:
worth switching the order of execution, or storing in mem the old sessionWindow?

Reply:
Agree. Done.